src/Access/ContextAccess.h

```diff
@@ -203,4 +206,112 @@ class ContextAccess : public std::enable_shared_from_this<ContextAccess>
     mutable std::shared_ptr<const EnabledSettings> enabled_settings TSA_GUARDED_BY(mutex);
 };
 
+
+class ContextAccessWrapper : public std::enable_shared_from_this<ContextAccessWrapper>
```

Comment:
Please add a comment explaining why we need the wrapper

Reply:
Done